### DISCUSSION TITLE
blocklist

### DISCUSSION THREAD

**Discussion by gregdenay** [bio-raum/FooDMe2#8] [General]
@marchoeppner 

HELPER_BLAST_APPLY_BLOCKLICT fails if no blocklist is provided.
An empty file should be provided or the step skipped. What is the Nexflow way?

**Comment by gregdenay** [comment]
Nevermind the blocklist is under asset but not passed on to the script

**Comment by marchoeppner** [comment]
Hi - sure that it isn't?

https://github.com/bio-raum/FooDMe2/blob/5d98fc9aebf5ffd4d653f5af3b406d20def4b216/nextflow.config#L26

This should set the built-in blocklist as default, here:

https://github.com/bio-raum/FooDMe2/blob/5d98fc9aebf5ffd4d653f5af3b406d20def4b216/workflows/foodme2.nf#L75

For me, the process directory looks as follows:

```
ls -lh  work/ab/0d8dc0ddeeaae239935ac0612d7e00/
insgesamt 76K
-rw-r--r--. 1 marc marc 69K 25. Jun 06:47 blast.blocked.mask
lrwxrwxrwx. 1 marc marc  82 25. Jun 06:47 blast.mask -> /home/marc/projects/Foodme2/test/work/7f/854bbf22d8137cbc3bf90d227cb7d4/blast.mask
lrwxrwxrwx. 1 marc marc  43 25. Jun 06:47 blocklist.txt -> /home/marc/git/FooDMe2/assets/blocklist.txt
-rw-r--r--. 1 marc marc  75 25. Jun 06:47 versions.yml
``` 

So basically, we use params.blocklist to set a default blocklist, but the option is user-editable - so a user could access it to overwrite the block list with their own list.

**Comment by marchoeppner** [comment]
As for Nextflow ways to deal with optional inputs - that depends on the exact situation. For this example, we could either pass an empty file (instead of the old default block list, as done here) so that technically nothing gets blocked but the process would still run, or we could also do:

```NEXTFLOW
if (params.blocklist) {
     HELPER_BLAST_APPLY_BLOCKLIST(
         blastmask,
         blocklist
      )
      ch_blocked_mask = HELPER_BLAST_APPLY_BLOCKLIST.out.mask
} else { 
     ch_blocked_mask = blastmask
}
```  

This assumes that the channel `blocklist` exists - but it would be empty if params.blocklist is not set so the process would "break" - hence why we put it in an if/else loop and otherwise set the default blast mask as the "blocked mask" (i.e. leave it unchanged).

**Comment by gregdenay** [comment]
ehm no it works as intended. Turned out I pulled the last commit with my VScode open and it overwrote some of the files...
